compiler/zrc_parser/src/parser.rs

```diff
@@ -125,9 +146,13 @@ pub fn parse_program(input: &str) -> Result<Vec<Spanned<Declaration<'_>>>, Diagn
 /// # Errors
 /// This function returns [`Err`] with a [`ZircoParserError`] if any error was
 /// encountered while parsing the input statement list.
+#[allow(clippy::result_large_err)]
 pub fn parse_stmt_list(input: &str) -> Result<Spanned<Vec<Stmt<'_>>>, Diagnostic> {
     internal_parser::StmtListParser::new()
-        .parse(lexer::ZircoLexer::new(input).map(zirco_lexer_span_to_lalrpop_span))
+        .parse(
+            "",
```

Comment:
All of these functions should REQUIRE a file name. In tests, `<source>` will be used. For stdin, `-` or `<stdin>` will be used.